### ISSUE TITLE
Unable to remove breakpoint with Blazor WASM while debugging

### ISSUE THREAD

**Issue by mkArtakMSFT** [mono/mono#19682]
_From @StevenRasmussen on Wednesday, April 29, 2020 4:57:44 PM_

When debugging a Blazor WASM project, if I set a breakpoint and then try to remove it while debugging, the breakpoint is still hit and execution stops even though the breakpoint has been "removed".

Environment:
Visual Studio 2019 Version 16.6.0 Preview 4.0
Blazor: v3.2.0-preview5.20216.8
Browser: MS Edge Version 81.0.416.64 (Official build) (64-bit)

_Copied from original issue: dotnet/aspnetcore#21328_

**Comment by baulig** [comment]
This is a known issue, I already have a fix for it, just didn't have time to make a PR for it.

**Comment by baulig** [comment]
We recently fixed this in https://github.com/mono/mono/pull/19578.

I also had the following in my local working tree: https://github.com/baulig/mono/commit/51628396a571539fbd3b8ff0bb72863d68ee31a7.

But it looks like it's not not needed, removing and reinserting breakpoints is working fine.

**Comment by StevenRasmussen** [comment]
@baulig - Awesome!  The debugging experience is pretty amazing all things considered ;). Thanks!

**Comment by baulig** [comment]
> I also had the following in my local working tree: [baulig@5162839](https://github.com/baulig/mono/commit/51628396a571539fbd3b8ff0bb72863d68ee31a7).
> 
> But it looks like it's not not needed, removing and reinserting breakpoints is working fine.

After having a closer look at that, we really don't need that commit from my local working tree, so I'm not going to make a PR for it.

So to summarize, this issue should be fixed by now.

**Comment by StevenRasmussen** [comment]
@baulig - Is there an updated package that I need to get or some configuration that needs to be updated?  I just updated all my packages to RC1 and am still experiencing this issue.

**Comment by StevenRasmussen** [comment]
@baulig - Interesting... this does appear to be working now.  Not sure what happened but I had closed VS a couple of times and rebuilt the solution as well... perhaps somewhere along the line it did what it needed to in order to start working ;)

**Comment by samsengupta** [comment]
I have experienced this issue from when I started using Blazor WASM years ago, and it still happens to this day in October 2025. I am using VS 2022 17.14.16. Breakpoints are hit in Blazor WASM, but while the app is running, breakpoint removal removes the red dot, but the breakpoints still get hit until I restart the app. All projects in my solution are using .NET 9, but the issue has consistently happened in prior versions of VS and .NET.
